tests/unit/test_syscontainers.py

```diff
@@ -291,6 +291,37 @@ def test_handle_system_package_files(self, _isi, _raf, _gr):
         rpm_install_output_third = sc._handle_system_package_files(test_options_two, manifest, "")
         self.assertEqual(rpm_install_output_third["new_installed_files_checksum"], expected_checksum)
 
+@unittest.skipIf(no_mock, "Mock not found")
+class TestSystemContainers_miscellaneous(unittest.TestCase):
+    """
+    Unit tests for general / utility functions in syscontainers.py
+    """
+    def test_encode_to_ostree_ref(self):
+        """
+        Verify that 1: image after encoded to ostree ref will be no longer hex
+        2: image contains no tag will have tag latest appended
+        3: qualified image(image with registry included) and non-qualified image will behave in the same way
+        when calling _encode_to_ostree_ref
+        """
+
+        qual_img_without_tag = "registry.fedoraproject.org/f27/kubernetes-apiserver"
+        qual_img_with_tag = qual_img_without_tag + ":tag"
+        non_qual_img_without_tag = "cafe"
+        non_qual_img_with_tag = non_qual_img_without_tag + ":tag"
+
+        qual_img_without_tag_ref = SystemContainers._encode_to_ostree_ref(qual_img_without_tag)
```

Comment:
let's check for `self.assertFalse(ref_is_hex)` for all the cases here.

Reply:
ah, the reason why I did not check other cases is that, the initial img itself is already non-hex. But yea, I will add back the check for other imgs as well. 

and btw, did you see my comment above regarding the other case?